commands/deploy_test.go

```diff
@@ -42,6 +49,62 @@ func Test_deploy(t *testing.T) {
 	}
 }
 
+func Test_oveerrideFlag_deployToNamespace(t *testing.T) {
```

Comment:
I was also thinking about how to test this, it might be easier to move `functionNamespace` to `DeployFlags.Namespace` (updating how the flag is bound) and then pulling out the code for creating the request into a 

```go
func PrepareDeployRequest(function stack.Function, flags DeployFlags) proxy.DeployFunctionSpec {}
```

which allows you to just test that the correct request is constructed from the flags, avoiding the need for the mock http server, capture stdout, and the tempfile.

The current test is still good but tests several things at once, flag binding, parsing, file reading, request construction, and the specific endpoint used

Reply:
Yes, @LucasRoesler I was wondering too about testing, because it is a bit challenging, also because you need to pull in the templates before you run the test, or it just fails. What you're suggesting seems really efficient, just check the faas-cli request!